angular-chat-app/src/app/room1/room1.component.ts

```diff
@@ -44,11 +45,24 @@ export class Room1Component implements OnInit, OnChanges {
         console.log(`Joined room: ${this.room}`);
       });
       this.name = '';
+      if (this.crud.userName !== '') {
+      this.name = this.crud.userName
```

Comment:
- store the connected user inside the localstorage instead of a hard variable.
- Also you should abstract the user check state from this component. Create a method isLogin() instead.
- Crud should be named AuthService